### DISCUSSION TITLE
Add dimension and unit hints to inputs and outputs

### DISCUSSION THREAD

**Discussion by GrimPixel** [IndiePython/nodezator#71] [Feature Request]
There is something called “dimension”, which consists of time (T), length (L), mass (M), electric current (I), absolute temperature (Θ), amount of substance (N), luminous intensity (J). They are useful for checking if one connected values of wrong dimensions.
In the same dimension, people may conduct mistakes by applying values of wrong units. This can be done by simply writing them as a part of the input and output names, or in a complicated way so the program can detect if units are wrong.

**Comment by KennedyRichard** [comment]
I think you mentioned this before, no? Could you please provide a link to some material explaining this or explain this further yourself? An example in Python would help as well if you could please provide one.

I think I understand what you are saying more or less, but I'm struggling to understand the role of the app in tackling this, since it might be something that the developer has to deal with in the code itself. Sorry if I'm not being clear enough, I'm struggling with this topic a bit.

**Comment by KennedyRichard** [comment]
I found the comment where you mentioned this. It is this one: https://github.com/IndiePython/nodezator/discussions/47#discussioncomment-6514567

I can't tackle this right now, but it is good that you created this discussion about the topic. After implementing the other higher priority features I'll visit this discussion again so we can resume the discussion on this topic. Meanwhile you or anyone can add more info/comment on it as desired/needed. For now, it is not a priority.

**Reply by GrimPixel** [reply]
Ah, yes. I said so. Some new ideas have come out, so I am not eager on this one anymore. Thank you for your efforts in finding that.
